pkg/virtctl/vmexport/vmexport.go

```diff
@@ -0,0 +1,639 @@
+						downloadUrl = format.Url
+						httpHeader = exportTokenHeader
+						break
+					} else if format.Format == exportv1.Dir {
+						downloadUrl = format.Url
+						httpHeader = dirExportTokenHeader
+						break
+					}
+				} else {
+					if format.Format == exportv1.KubeVirtGz || format.Format == exportv1.ArchiveGz {
+						downloadUrl = format.Url
+						httpHeader = exportTokenHeader
+						break
+					}
+				}
+			}
+		}
+	}
+
+	if downloadUrl == "" {
+		return "", "", fmt.Errorf("Unable to get a valid URL from '%s/%s' VirtualMachineExport", vmexport.Namespace, vmexport.Name)
+	}
+
+	return downloadUrl, httpHeader, nil
+}
+
+// waitForVirtualMachineExport waits for the VirtualMachineExport status to be ready
+func waitForVirtualMachineExport(client kubecli.KubevirtClient, namespace string, interval, timeout time.Duration) error {
+	err := wait.PollImmediate(interval, timeout, func() (bool, error) {
+		vmexport, err := getVirtualMachineExport(client, namespace)
+		if err != nil {
+			return false, err
+		}
+
+		if vmexport == nil || vmexport.Status == nil {
+			return false, nil
+		}
+
+		done := vmexport.Status.Phase == "Ready"
+		if !done {
+			fmt.Printf("Waiting for VM Export %s status to be ready...\n", vmexportName)
+			return false, nil
+		}
+
+		if vmexport.Status.Links == nil || vmexport.Status.Links.External == nil {
+			fmt.Printf("Waiting for VM Export %s external links to be available...\n", vmexportName)
+			return false, nil
+		}
+
+		fmt.Printf("Processing completed successfully\n")
+		return done, nil
+	})
+
+	return err
+}
+
+// handleHTTPRequest generates the GET request with proper certificate handling
+func handleHTTPRequest(client kubecli.KubevirtClient, vmexport *exportv1.VirtualMachineExport, downloadUrl, httpHeader string) (*http.Response, error) {
+	token, err := getTokenFromSecret(client, vmexport.Namespace)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create new certPool and append our external SSL certificate
+	cert := vmexport.Status.Links.External.Cert
+	roots := x509.NewCertPool()
+	roots.AppendCertsFromPEM([]byte(cert))
+	transport := &http.Transport{
+		TLSClientConfig: &tls.Config{RootCAs: roots},
+	}
+	httpClient := httpClientCreatorFunc(transport)
+
+	// Generate and do the request
+	req, _ := http.NewRequest("GET", downloadUrl, nil)
+	req.Header.Set(httpHeader, token)
+
+	return httpClient.Do(req)
+}
+
+// getHTTPClient assigns the default, non-mocked HTTP client
+func getHTTPClient(transport *http.Transport) *http.Client {
+	if transport == nil {
+		transport = &http.Transport{
+			TLSClientConfig: &tls.Config{
+				InsecureSkipVerify: true,
+			},
+		}
+	}
+
+	client := &http.Client{Transport: transport}
+	return client
+}
+
+// copyFileWithProgressBar serves as a wrapper to copy the file with a progress bar
+func copyFileWithProgressBar(output *os.File, resp *http.Response) error {
+	// TODO: Unable to get file size when downloading using the compressed URL
```

Comment:
If you update the progress bar to [3.1.0](https://github.com/cheggaaa/pb/releases/tag/v3.1.0) you can provide a template that has a repeating pattern, which is essentially a spinner. You can use this for the case you are unable to determine